### ISSUE TITLE
Results on mono-data not good

### ISSUE THREAD

**Issue by Heimine** [nianticlabs/monodepth2#171]
Hi guys, thanks for the wonderful work! I have a question when I use a custom mono dataset, which is like this: 
![image](https://user-images.githubusercontent.com/43052323/80328469-71000300-880d-11ea-8dbe-808770e8d016.png)
There are six cameras from different angle of the car. 
I tried to train the model from scratch using the mono-way (choose an image and the images immediately before and after this image). But the result is not good. At it best (4000 random images as training dataset), I got this: 
![image](https://user-images.githubusercontent.com/43052323/80328649-ebc91e00-880d-11ea-82f9-6e380ff5e442.png)
At first I thought maybe because I use images of all cameras so the performance is been compromised, so I turn to use images all from one camera as training set (again 4000 images), but the result got worse... Like this:
![image](https://user-images.githubusercontent.com/43052323/80328721-2763e800-880e-11ea-878b-5bf85f85873c.png)
So I'm wondering could you please give me any suggestions to get better results? 
Thank you very much!

**Comment by mrharicot** [comment]
Hi,
It is difficult to say what might be the problem here.
4000 is also a very small number of training samples.
Do all cameras have the same focal length? And did you modify the intrinsics matrix accordingly?
What is the framerate?
Can you share a few example triplet frames?

**Comment by Heimine** [comment]
> Hi,
> It is difficult to say what might be the problem here.
> 4000 is also a very small number of training samples.
> Do all cameras have the same focal length? And did you modify the intrinsics matrix accordingly?
> What is the framerate?
> Can you share a few example triplet frames?

Hi thanks for the quick reply!
Each camera has different intrinsics and I make sure that the model use the right one for each input image. and there is about 0.2 seconds between consecutive samples. Here's 2 sets of example triplet:
![image](https://user-images.githubusercontent.com/43052323/80334269-b678fc00-881e-11ea-83ae-9805023fbf54.png)
![image](https://user-images.githubusercontent.com/43052323/80334285-c1339100-881e-11ea-9081-f8ad9101a750.png)
![image](https://user-images.githubusercontent.com/43052323/80334302-c98bcc00-881e-11ea-825a-32180abde3e0.png)

![image](https://user-images.githubusercontent.com/43052323/80334349-f63fe380-881e-11ea-8bef-7791706ad2c1.png)
![image](https://user-images.githubusercontent.com/43052323/80334362-fe981e80-881e-11ea-9c2b-fdd978cf36a0.png)
![image](https://user-images.githubusercontent.com/43052323/80334370-08218680-881f-11ea-93f9-2acab34de51a.png)

Thank you very much!

**Comment by mrharicot** [comment]
Thanks for the images, they seem fine except the low resolution and high compression.
Can you obtain higher quality ones for training?
Do you mind sharing the code of your dataloader?
Finally I would suggest trying with only the forward facing cameras for now, as an experiment to try and simplify the task to debug.

**Comment by Heimine** [comment]
Hi thanks again for the quick reply!
Actually I can't get higher quality images...
I tried to use only the front camera and got this result:
![image](https://user-images.githubusercontent.com/43052323/80336418-32764280-8825-11ea-9cee-73b58407c415.png)

Here's the code for dataset.
https://github.com/Heimine/monodepth2/blob/b16099f7db0c3062845b566eb9557dfb14e1a544/datasets/DL_dataset.py

Thanks so much for helping!

**Comment by mrharicot** [comment]
You need to normalize the intrinsics by image_width (first row) and image_height (second row)

**Comment by Heimine** [comment]
> You need to normalize the intrinsics by image_width (first row) and image_height (second row)

Thanks a lot! I'll fix that!

**Comment by Heimine** [comment]
It works! Thanks so much for helping!

**Comment by mrharicot** [comment]
Great! Do you mind sharing a predicted depthmap? Just for curiosity :)

**Comment by Heimine** [comment]
> Great! Do you mind sharing a predicted depthmap? Just for curiosity :)

Haha for sure!
![image](https://user-images.githubusercontent.com/43052323/80453775-ebf11880-88f6-11ea-8d0c-e5d57d80aaf6.png)

It's not perfect but I think it's due to the image quality. :)
Thank you very much for being so patient and helpful!

**Comment by karanchahal** [comment]
Thats awesome @Heimine , are you taking part in the NYU Deep learning competition as well haha ? I had the same exact doubt.
